src/MissionManager/MissionController.cc

```diff
@@ -1033,8 +1036,9 @@ void MissionController::_recalcWaypointLines(void)
         // If we still haven't found the first coordinate item and we hit a takeoff command, link back to home
         if (firstCoordinateItem &&
                 item->isSimpleItem() &&
-                (qobject_cast<SimpleMissionItem*>(item)->command() == MavlinkQmlSingleton::MAV_CMD_NAV_TAKEOFF ||
-                 qobject_cast<SimpleMissionItem*>(item)->command() == MavlinkQmlSingleton::MAV_CMD_NAV_VTOL_TAKEOFF)) {
+                (!_controllerVehicle->firmwarePlugin()->supportedMissionCommands().contains(MAV_CMD_NAV_TAKEOFF) ||
```

Comment:
This part should not change. This is testing whether a takeoff command is already in the mission.

Reply:
without this, when planning the mission, the home do not connect with the first item.

Reply:
Ok, I get it now. I wasn't understanding the code correctly. This is right.